packages/react-components/react-alert/src/components/Alert/Alert.types.ts

```diff
@@ -1,15 +1,27 @@
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { Button } from '@fluentui/react-button';
 
 export type AlertSlots = {
-  root: Slot<'div'>;
+  root: NonNullable<Slot<'div'>>;
+  icon?: Slot<'span'>;
+  content: NonNullable<Slot<'span'>>;
+  action?: Slot<typeof Button>;
+};
+
+type AlertCommons = {
+  /**
+   * The intent for the alert
+   */
+  // intent?: 'info' | 'success' | 'error' | 'warning' | 'infoAvatar';
+  intent?: 'info' | 'success' | 'error' | 'warning';
 };
 
 /**
  * Alert Props
  */
-export type AlertProps = ComponentProps<AlertSlots>;
+export type AlertProps = ComponentProps<AlertSlots> & AlertCommons;
```

Comment:
We're moving away from `Common` types (link to issue [here](https://github.com/microsoft/fluentui/issues/22862). You should inline the props here and `Pick` in state.

```suggestion
export type AlertProps = ComponentProps<AlertSlots> & {
  /**
   * The intent for the alert
   */
  intent?: 'info' | 'success' | 'error' | 'warning';
};
```